### ISSUE TITLE
NVDA not reading Headers/Footers

### ISSUE THREAD

**Issue by J0nSnuw** [nvaccess/nvda#18148]
### Steps to reproduce:

Using the attached PDF, viewing in Adobe Acrobat and NVDA while pressing (NVDA+F7), none of the tagged headers and paragraphs are loaded.

[ForNVDASubmissions.pdf](https://github.com/user-attachments/files/20407087/ForNVDASubmissions.pdf)

When I open the same PDF using FireFox, tags and headers load correctly, which seems to implies sufficient information exists on the PDF to make it navigable.

### Actual behavior:

Opening Adobe Acrobat with the attached PDF, pressing NVDA+F7 results in no headers loaded.

### Expected behavior:
Example using Firefox to view the PDF, pressing NVDA+F7, headers are loaded correctly based of the PDF tags.

#### NVDA version:

2024.4.2.35031

#### Windows version:

Windows 11

#### Adobe Acrobat Pro version:

2025.001.20474 | 64-bit

#### Firefox version:

138.0.3 (64-bit)

#### Other information about your system:

The PDF “ForNVDASubmissions” passes PAC24’s PDF/UA and WCAG 2.2 tests. It does not pass the quality check because it doesn’t have bookmarks

**Comment by seanbudd** [comment]
Welcome @J0nSnuw,

We are unable to process this issue as it stands.

- We need the source of the PDF to understand how it is generated and to confirm the tags. Can you please provide the steps you followed to generate the PDF?
- The testing that passes doesn't confirm the existence of these tags. It is possible that firefox is making assumptions for tagging in the absence of proper tags.

Kind Regards,
NV Access Software Developers

**Comment by J0nSnuw** [comment]
Hello @seanbudd 

Sorry for the delay.

My company is using proprietary off-the-shelf software PDF4Net from this [website](https://www.o2sol.com/pdf4net/overview.htm).

To try and confirm this PDF has tags, I have opened it using the free version of PDF-XChange, and I see the attached list of tags:

![Image](https://github.com/user-attachments/assets/a9cfb89a-f91f-4844-ad93-f241326ebfe5)

I cannot provide instructions on how to recreate it, and I do not expect NVDA or anyone else to purchase a license. However, I can confirm (or it appears so) that the PDF is tagged.

Feel free to let me know if there is anything else I can provide,
J0nSnuww

**Comment by seanbudd** [comment]
Thanks, we might have to contact the software authors for more information, but this should be enough to help

**Comment by SaschaCowley** [comment]
Interesting, [PAC](https://pac.pdf-accessibility.org/en) doesn't seem to show any major problems with it either. It may be because there aren't bookmarks, but I wouldn't think that's the case

**Comment by davidcarlisle** [comment]
The problem is that the H1 and H2 have /ActualText attributes this causes the entire element to be replaced by the specified text so it's not really a header, if I remove the attributes acrobat announces the header and stepping through with `h` works

[uncompressed-ForNVDASubmissions.pdf](https://github.com/user-attachments/files/20646142/uncompressed-ForNVDASubmissions.pdf)

**Comment by J0nSnuw** [comment]
Thanks David, I will look into this in our internal software and get back
to you.

I hope that is all that is needed and should be an easy fix!
